src/Microsoft.SqlTools.ServiceLayer/MachineLearningServices/ServcieOperations.cs

```diff
@@ -0,0 +1,116 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using System;
+using System.Data;
+
+namespace Microsoft.SqlTools.ServiceLayer.MachineLearningServices
+{
+    public class MachineLearningServcieOperations
+    {
+        private const int ExternalScriptConfigNumber = 1586;
+        private const string LanguageStatusScript = @"SELECT is_installed
+FROM sys.dm_db_external_language_stats s, sys.external_languages l
+WHERE s.external_language_id = l.external_language_id AND language = @LanguageName";
+
+        /// <summary>
+        /// Updates external script in a given server. Throws exception if server doesn't support external script 
+        /// </summary>
+        /// <param name="serverConnection"></param>
+        /// <param name="configValue"></param>
+        public void UpdateExternalScriptConfig(ServerConnection serverConnection, bool configValue)
+        {
+            Server server = new Server(serverConnection);
+            ConfigProperty serverConfig = GetExternalScriptConfig(server);
+
+            if (serverConfig != null)
+            {
+                try
+                {
+                    serverConfig.ConfigValue = configValue ? 1 : 0;
+                    server.Configuration.Alter(true);
+                }
+                catch (FailedOperationException ex)
+                {
+                    throw new MachineLearningServicesException("Failed to update external script config", ex);
```

Comment:
Do these exception messages need to be localized?

Reply:
the errors from servers are not localized I think but will double check

---
In reply to: [364377582](https://github.com/microsoft/sqltoolsservice/pull/906#discussion_r364377582) [](ancestors = 364377582)